compose-Satosa-Saml2Spid/nginx/nginx.conf_proxy_http

```diff
@@ -0,0 +1,89 @@
+user nginx;
+worker_processes  auto;
+pid /run/nginx.pid;
+
+events {
+    worker_connections  1024;
+}
+
+http {
+    include       mime.types;
+    default_type  application/octet-stream;
+
+    sendfile        on;
+    keepalive_timeout  240s;
+    send_timeout 240s;
+    client_max_body_size 3300m;
+
+    log_format main '$http_x_forwarded_for - $remote_user [$time_local] '
+                    '"$request" $status $body_bytes_sent "$http_referer" '
+                    '"$http_user_agent"' ;
+
+# the upstream component nginx needs to connect to
+#upstream satosa-saml2 {
+#    server satosa-saml2spid:10000;
+#}
+
+# configuration of the server
+server {
+  listen      80;
+  server_name proxy.local;
+  access_log /var/log/nginx/access.log;
+  error_log  /var/log/nginx/error.log error;
+  return 301 https://$host$request_uri;
+}
+
+server {
+    server_name proxy.local;
+    listen 443 ssl;
+
+    ssl_certificate /etc/nginx/certs/proxy_local.pem;
+    ssl_certificate_key /etc/nginx/certs/proxy_local.key;
+
+    access_log /var/log/nginx/access.log;
+    error_log  /var/log/nginx/error.log error;
+
+    # max upload size
+    client_max_body_size 8M;
+
```

Comment:
may we have a common hardening strategy like this?
https://github.com/UniversitaDellaCalabria/uniTicket/blob/master/uwsgi_setup/nginx#L45